index.js

```diff
@@ -2,30 +2,67 @@ import execa from 'execa';
 import pMemoize from 'p-memoize';
 import delay from 'delay';
 
+const powershell = (async command => {
+	if (typeof (command) !== 'string') {
+		throw new TypeError('Invalid command');
+	}
+
+	return execa.command(command, {shell: 'powershell'});
+});
+
 const getDevice = pMemoize(async () => {
-	if (process.platform !== 'darwin') {
-		throw new Error('macOS only');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+		const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result.groups.device;
 	}
 
-	const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapter -Name Wi-Fi | Format-List -Property PnPDeviceID');
+		const result = /(?<=PnPDeviceID : ).*/.exec(stdout);
+
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
 
-	const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
-	if (!result) {
-		throw new Error('Couldn\'t find a Wi-Fi device');
+		return result[0];
 	}
 
-	return result.groups.device;
+	throw new Error('macOS or Windows only');
 });
 
 const isOn = async device => {
-	const {stdout} = await execa('ifconfig', [device]);
-	return stdout.includes('status: active');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('ifconfig', [device]);
+		return stdout.includes('status: active');
+	}
+
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapterAdvancedProperty -Name Wi-Fi -RegistryKeyword RFOff -AllProperties | Format-List -Property RegistryValue');
+		const status = /(?<={).+?(?=})/.exec(stdout);
```

Comment:
Add a comment with an example of what `stdout` is. That makes it easier to read the code and interpret the regex. Same in other places.